crates/ln-dlc-node/src/ln/coordinator_event_handler.rs

```diff
@@ -345,14 +355,31 @@ fn handle_open_channel_request<S: TenTenOneStorage, N: Storage>(
     funding_satoshis: u64,
     push_msat: u64,
     temporary_channel_id: [u8; 32],
+    channel_type: ChannelTypeFeatures,
 ) -> Result<()> {
     let counterparty = counterparty_node_id.to_string();
     tracing::info!(
         counterparty,
         funding_satoshis,
         push_msat,
-        "Accepting open channel request"
+        "Handling open channel request"
     );
+
+    if channel_type.requires_zero_conf() {
+        bail!("Rejecting open channel request. Reason: The coordinator does not accept zero conf inbound channels");
+    }
+
+    if channel_type.requires_anchors_zero_fee_htlc_tx() {
+        let counterparty_reserve = channel_manager
+            .get_channel_details(&temporary_channel_id)
+            .context("Couldn't find channel by temporary channel id")?
+            .counterparty
+            .unspendable_punishment_reserve;
+        if counterparty_reserve < MIN_COUNTERPARTY_RESERVE_SATS {
+            bail!("Rejecting open channel request. Reason: The unspendable punishement reserve is too small. Min: {MIN_COUNTERPARTY_RESERVE_SATS}, Got: {counterparty_reserve}")
```

Comment:
🔧 Also we could split the string into two so that it doesn't go over the usual column width of 100 characters.

```suggestion
            bail!("Rejecting open channel request. Reason: The unspendable counterparty punishment reserve is too small. Min {MIN_COUNTERPARTY_RESERVE_SATS} sats, got {counterparty_reserve} sats")
```